Brightness recognition

BalzPit opened the pull request:
Added a method "IMG_Brightness" that determines wether or not the image is too dark or bright.
Before calling the intent at the end of TakePhoto in CameraActivity the method is called and if it detects over or underexposure a Toast will pop up notifying the user

BalzPit committed:
First prototype of image brightness recognition

BalzPit committed:
Tuning, Added notification to the user

if the image is either too dark or bright a Toast will now pop up informing the user that the text might not be recognized properly

PrandiniUniPD reviewed (commented):
I've tested this app. I've received the toast in the ResultActivity after some time because the app waits the result of the recognition before continuing with the ResultActivity.
Is it a good idea running this recognition to another thread in the ResultActivity?
If not, is it a good idea running the recognition in an acynctask to notify the user the app are working to him?
For too bright pictures I've test a photo of a light is it right? (toast not received)
For too dark picture I've test a photo of a no light environment is it right? (toast received)

![photo_2018-11-27_11-16-16](https://user-images.githubusercontent.com/44234534/49074899-ff95dc80-f235-11e8-8179-08f59c7dd243.jpg)
![photo_2018-11-27_11-16-10](https://user-images.githubusercontent.com/44234534/49075228-b003e080-f236-11e8-9eb0-95cd1fccdef6.jpg)

BalzPit committed:
Performance Improvement

Created the class BrightnessRecognition to run the recognition on a separate thread to improve performance. The class is used in "cameraActivity", after the thread is opened

frankplus committed:
Renamed to comply style document

frankplus committed:
Some performance improvements

frankplus committed:
Some performance improvements

frankplus reviewed (approved):
Looks good to me

frankplus committed:
Some changes

BalzPit committed:
minor changes

BalzPit committed:
ready for code review

PrandiniUniPD commented on app/src/main/java/unipd/se18/ocrcamera/CameraActivity.java:
```diff
@@ -156,6 +163,17 @@ public void onImage(CameraKitView cameraKitView, final byte[] photo) {
 
+                    // analyze the brightness   @author Balzan Pietro
+                    final Bitmap bitmapImage2 = bitmapImage;
+                    new Thread(new Runnable() {      // run the brightness recognition on a new thread to improve performance
```
Is it really useful running these statements in a new Thread considered that the code of the BrightnessRecognition is in another java class and so in another Thread?

BalzPit replied:
Maybe i coded it wrong but when I tried just running the statements like you said, the performance wasn't improved: it still waited for the Recognition to end before launching the ResultActivity with the intent. It was like there wasn't a new automatic thread, so I made it manually.

PrandiniUniPD commented on app/src/main/java/unipd/se18/ocrcamera/BrightnessRecognition.java:
```diff
@@ -0,0 +1,93 @@
+     * @author Pietro Balzan - some changes by Francesco Pham
+     */
+
+    public void imgBrightness(int upperBound, int lowerBound, int pixelSkip){
```
Could be useful if this method returns an int for a possible future automatic processing of the picture? For example -1 too dark, 0 ok, 1 too bright?
Is really useful setting externally these parameters? Or they could be calculated automatically or anyway internally to this method?

BalzPit replied:
good idea, I will add that feature

PrandiniUniPD commented on app/src/main/java/unipd/se18/ocrcamera/BrightnessRecognition.java:
```diff
@@ -0,0 +1,93 @@
+     * @author Pietro Balzan
+     */
+
+    public BrightnessRecognition(Bitmap bmp, Context cntxt, Handler hndlr){
```
Could have sense if this class contains only static methods without a constructor?

BalzPit replied:
It makes sense

PrandiniUniPD reviewed (commented):
I prefer this version with the processing in another class, good work.

BalzPit committed:
Merge branch 'CommonDemo' into Brightness_recognition

BalzPit committed:
Suggestions added to the code

added integer return value to imgBrightness and moved the toasts to be called inside the thread in takePhoto method of CameraActivity

BalzPit committed:
Removed constructor and added static method implementation

Nintenking reviewed (approved):
Works fine for me, could you now do a Pull request to the branch "ImagePorcessing_module"? We're making a dedicated module and putting there all the Image processing features

frankplus commented:
Just for the record, I tested the time of execution of imgBrightness(), on my phone (Mi A2) with pixelSkip=3 it takes about 130 ms, it is ok to me.

Fisher4537 commented:
Why is this pull request already open? @BalzPit is it alrready active? We discussed about the class which should countain all the methods of image manipulation! Do you change your mind?

Nintenking commented:
This pull request was made before our idea of the Image Processing module, and we already merged this branch in the one with the module. He surely just forgot to close this pull request, so I'm doing it now.
